sdk/translation/Azure.AI.Translation.Document/src/BatchDocumentTranslationModelFactory.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        /// <param name="sourceDocumentUri">Sets the <see cref="DocumentStatus.SourceDocumentUri"/> property.</param>
+        /// <param name="createdOn">Sets the <see cref="DocumentStatus.CreatedOn"/> property.</param>
+        /// <param name="lastModified">Sets the <see cref="DocumentStatus.LastModified"/> property.</param>
+        /// <param name="status">Sets the <see cref="DocumentStatus.Status"/> property.</param>
+        /// <param name="translatedTo">Sets the <see cref="DocumentStatus.TranslatedTo"/> property.</param>
+        /// <param name="progress">Sets the <see cref="DocumentStatus.Progress"/> property.</param>
+        /// <param name="id">Sets the <see cref="DocumentStatus.Id"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.</returns>
+        /// /// <exception cref="ArgumentNullException"> <paramref name="sourceDocumentUri"/>, <paramref name="translatedTo"/>, or <paramref name="id"/> is null. </exception>
+        public static DocumentStatus DocumentStatus(
+            Uri sourceDocumentUri,
+            DateTimeOffset createdOn,
+            DateTimeOffset lastModified,
+            DocumentTranslationStatus status,
+            string translatedTo,
+            float progress,
+            string id
+            )
+        {
+            return new DocumentStatus(default, sourceDocumentUri, createdOn, lastModified, status, translatedTo,default, progress, id, default);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.
+        /// </summary>
+        /// <param name="translatedDocumentUri">Sets the <see cref="DocumentStatus.TranslatedDocumentUri"/> property.</param>
+        /// <param name="sourceDocumentUri">Sets the <see cref="DocumentStatus.SourceDocumentUri"/> property.</param>
+        /// <param name="createdOn">Sets the <see cref="DocumentStatus.CreatedOn"/> property.</param>
+        /// <param name="lastModified">Sets the <see cref="DocumentStatus.LastModified"/> property.</param>
+        /// <param name="status">Sets the <see cref="DocumentStatus.Status"/> property.</param>
+        /// <param name="translatedTo">Sets the <see cref="DocumentStatus.TranslatedTo"/> property.</param>
+        /// <param name="error">Sets the <see cref="DocumentStatus.Error"/> property.</param>
+        /// <param name="progress">Sets the <see cref="DocumentStatus.Progress"/> property.</param>
+        /// <param name="id">Sets the <see cref="DocumentStatus.Id"/> property.</param>
+        /// <param name="charactersCharged">Sets the <see cref="DocumentStatus.CharactersCharged"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.</returns>
+        public static DocumentStatus DocumentStatus(
```

Comment:
Side comment, @AlexanderSher could you explain to me why codegen didn't generate this method?